File: minefinder-web/src/services/mqtt/topics.ts
/**
 * MQTT topic constants for MineFinder system
 */

export class MQTTTopics {
  static systemStatus(): string {
    return 'minefinder/system/status';
  }

  static systemConfig(): string {
    return 'minefinder/system/config';
  }

  static attachmentStatus(attachmentId: string): string {
    return `minefinder/attachment/${attachmentId}/status`;
  }

  static attachmentHeartbeat(attachmentId: string): string {
    return `minefinder/attachment/${attachmentId}/heartbeat`;
  }

  static attachmentTelemetry(attachmentId: string): string {
    return `minefinder/attachment/${attachmentId}/telemetry`;
  }

  static attachmentDetection(attachmentId: string): string {
    return `minefinder/attachment/${attachmentId}/detection`;
  }

  static attachmentCommand(attachmentId: string): string {
    return `minefinder/attachment/${attachmentId}/command`;
  }

  static attachmentCommandAck(attachmentId: string): string {
    return `minefinder/attachment/${attachmentId}/command/ack`;
  }

  static missionStart(missionId: string): string {
    return `minefinder/mission/${missionId}/start`;
  }

  static missionStop(missionId: string): string {
    return `minefinder/mission/${missionId}/stop`;
  }

  static missionStatus(missionId: string): string {
    return `minefinder/mission/${missionId}/status`;
  }

  static missionPath(missionId: string): string {
    return `minefinder/mission/${missionId}/path`;
  }

  static missionEvents(missionId: string): string {
    return `minefinder/mission/${missionId}/events`;
  }
}

File: PathFinder/mqtt/topics.py
"""MQTT topic constants for MineFinder system"""

class MQTTTopics:
    """Topic structure for MQTT communication"""
    
    @staticmethod
    def system_status() -> str:
        return "minefinder/system/status"
    
    @staticmethod
    def system_config() -> str:
        return "minefinder/system/config"
    
    @staticmethod
    def attachment_status(attachment_id: str) -> str:
        return f"minefinder/attachment/{attachment_id}/status"
    
    @staticmethod
    def attachment_heartbeat(attachment_id: str) -> str:
        return f"minefinder/attachment/{attachment_id}/heartbeat"
    
    @staticmethod
    def attachment_telemetry(attachment_id: str) -> str:
        return f"minefinder/attachment/{attachment_id}/telemetry"
    
    @staticmethod
    def attachment_detection(attachment_id: str) -> str:
        return f"minefinder/attachment/{attachment_id}/detection"
    
    @staticmethod
    def attachment_command(attachment_id: str) -> str:
        return f"minefinder/attachment/{attachment_id}/command"
    
    @staticmethod
    def attachment_command_ack(attachment_id: str) -> str:
        return f"minefinder/attachment/{attachment_id}/command/ack"
    
    @staticmethod
    def mission_start(mission_id: str) -> str:
        return f"minefinder/mission/{mission_id}/start"
    
    @staticmethod
    def mission_stop(mission_id: str) -> str:
        return f"minefinder/mission/{mission_id}/stop"
    
    @staticmethod
    def mission_status(mission_id: str) -> str:
        return f"minefinder/mission/{mission_id}/status"
    
    @staticmethod
    def mission_path(mission_id: str) -> str:
        return f"minefinder/mission/{mission_id}/path"
    
    @staticmethod
    def mission_events(mission_id: str) -> str:
        return f"minefinder/mission/{mission_id}/events"
